### PR TITLE
Fixed all walletTest threads not being closed

### PR DESCRIPTION
**Pull request by nkohen**

### CODE REVIEW

**Review by benthecarman — APPROVED** [review]

File: testkit/src/main/scala/org/bitcoins/testkit/wallet/BitcoinSWalletTest.scala [hunk 1]
```diff
@@ -439,6 +440,7 @@ object BitcoinSWalletTest extends WalletLogger {
       wallet <- BitcoinSWalletTest.createWallet2Accounts(bitcoind,
                                                          bitcoind,
                                                          extraConfig)
+      _ = wallet.stopWalletThread()
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/testkit/wallet/BitcoinSWalletTest.scala:443]: Why is this here? Shouldn't this be in `destroyWallet`?

Comment (benthecarman) [reply testkit/src/main/scala/org/bitcoins/testkit/wallet/BitcoinSWalletTest.scala:443]: We create the wallet again so it can have callbacks after this, so this wallet will never passed to destroy wallet

Comment (Christewart) [reply testkit/src/main/scala/org/bitcoins/testkit/wallet/BitcoinSWalletTest.scala:443]: Ugh, so it seems this thread management should really happen in `WalletAppConfig` as well then? It seems like it is _so_ easy to forget to do this when a wallet is destroyed
